### DISCUSSION TITLE
Json string to json path

### DISCUSSION THREAD

**Discussion by rj10rd** [arendst/Tasmota#13922] [Q&A] [answered]
{"StatusSNS":{"Time":"2021-12-05T00:24:51","DS18B20":{"Id":"3C01D0754A0E","Temperature":25.4},"TempUnit":"C"}}

In above json string I Want to extract temperature with temperature unit like "25.4 C" . Please help me with it.

**Comment by barbudor** [comment, marked as answer]
I would assume you want to do that in a rule?
On StatusSNS#Temperature Do var1 %value% endon on StatusSNS#TempUnit do var2 %var1% %value% endon

**Reply by rj10rd** [reply]
No sir I want that in mqtt subscription.

**Reply by barbudor** [reply]
So that's dépendant of your backend, not tasmota

**Reply by sfromis** [reply]
In easy to use tools, it could be something like JSON parsing automatically creating variables/objects, allowing a notation like msg.payload.StatusSNS.DS18B20.Temperature in Node-Red. You may have selected a backend making it more difficult :-)

**Reply by rj10rd** [reply]
Please try this https://github.com/xDWart/mqtg-bot or https://telegram.me/mqtg_bot
And tell me how extract temperature with unit from json string

**Reply by sfromis** [reply]
Without posting a JSON string, and without info about where you want to extract temperature, do not expect a suggestion. If outside Tasmota, it is off topic for the project.

**Reply by rj10rd** [reply]
Sir i mension json string above and i wan temperature with unit in telegram mqtt client. Please check my given link

**Reply by barbudor** [reply]
If you want to use the Tasmota Telegram feature, we can support you
Now if you are using another software like mqtt-to-telegram, you should ask support at this other software how to configure it.

**Reply by sfromis** [reply]
What I meant was a specific example, instead of expecting someone else to dig through those pages, and guess at what you mean. Anyway, by now it sounds like your're asking the wrong place. We do not provide general help with JSON parsing here.
